### PR TITLE
Fix Latex Formula Parsing and Regex Simplification

### PR DESCRIPTION
**Pull request by pzierahn**
This pull request resolves a critical bug that previously prevented latex formulas from being parsed correctly. It also simplifies the regex statements used in the parsing process and cleans up various parts of the code for better readability and performance. Specific changes include:

1. **Latex Parsing Fix:** Adjusted the code to ensure latex formulas are correctly parsed, improving functionality for users dealing with mathematical content.

2. **Regex Simplification:** The complexity of regular expression statements for parsing has been reduced, making the code easier to maintain and more efficient.

3. **Code Cleanup:** Conducted a thorough cleanup of the codebase, removing unnecessary code, optimizing functions, and enhancing documentation for clarity and maintenance ease.

These enhancements serve to make latex formula handling more reliable and the codebase more accessible for future development. Feedback and further suggestions are welcome.

### CODE REVIEW

File: lib/src/latex_inline_syntax.dart [hunk 1]
```diff
@@ -1,17 +1,26 @@
...
+      return false;
+    }
+
+    final equation = (match[2] ?? '').trim();
+    if (equation.isEmpty) {
+      return false;
+    }
+
+    final element = Element.text('latex', equation);
+    element.attributes['MathStyle'] = 'text';
```
Comment (xushengs) [inline lib/src/latex_inline_syntax.dart]: Could you keep MathStyle to display mode when delimiter is $$?

**Comment by xushengs** [comment]
Thanks for your contribution!

File: example/pubspec.yaml
Comment (pzierahn) [inline example/pubspec.yaml:1]: I have accidentally appended commit to this pull request. The inclusion of the flutter_markdown library from my GitHub account could cause problems when publishing this package to pub.dev.

File: pubspec.yaml
Comment (pzierahn) [inline pubspec.yaml:1]: Same goes for this pubspec.yaml. Im working on fixing unwanted line-breaks in Markdown

**Comment by pzierahn** [comment]
I have created a new pull request #9 to fix it!

**Comment by xushengs** [comment]
The code doesn't seem to pass this test: 
```
  'inline latex 3 \$': '''
This is not inline latex: \$\$\$c = \\pm\\sqrt{a^2 + b^2}\$\$\$
''',
```

**Pull request merged**